Core/src/main/java/com/github/intellectualsites/plotsquared/plot/util/block/BasicLocalBlockQueue.java

```diff
@@ -116,32 +118,18 @@ public final boolean execute(final LocalChunk<T> lc) {
     }
 
     @Override public boolean setBlock(int x, int y, int z, PlotBlock id) {
-        if (y > 255) {
-            return false;
-        } else if (y < 0) {
-            return false;
-        }
-        int cx = x >> 4;
-        int cz = z >> 4;
-        if (cx != lastX || cz != lastZ) {
-            lastX = cx;
-            lastZ = cz;
-            long pair = (long) (cx) << 32 | (cz) & 0xFFFFFFFFL;
-            lastWrappedChunk = this.blockChunks.get(pair);
-            if (lastWrappedChunk == null) {
-                lastWrappedChunk = this.getLocalChunk(x >> 4, z >> 4);
-                lastWrappedChunk.setBlock(x & 15, y, z & 15, id);
-                LocalChunk previous = this.blockChunks.put(pair, lastWrappedChunk);
-                if (previous == null) {
-                    chunks.add(lastWrappedChunk);
-                    return true;
-                }
-                this.blockChunks.put(pair, previous);
-                lastWrappedChunk = previous;
-            }
+        // Trying to mix PlotBlock and BaseBlock leads to all kinds of issues.
+        // Since BaseBlock has more features than PlotBlock, simply convert
+        // all PlotBlocks to BaseBlocks
+        if (id instanceof StringPlotBlock) {
+            StringPlotBlock stringPlotBlock = (StringPlotBlock) id;
```

Comment:
It might be worth checking for (and storing) the cached base block in `StringPlotBlock` (https://github.com/IntellectualSites/PlotSquared/blob/41299e685156311d74661fb9aefa90325dcd47c8/Core/src/main/java/com/github/intellectualsites/plotsquared/plot/object/StringPlotBlock.java#L18) 

This is stored initially when getting a StringPlotBlock from a base block.

Reply:
I'm honestly a little embarrassed about how I did this using `instanceof` and down-casting rather than the more sensible abstract method approach. With the new approach in my latest commit, both StringPlotBlock and LegacyPlotBlock caches BaseBlock lookups.